### PR TITLE
EL-713: Start using postgres to record analytics

### PR DESCRIPTION
**Pull request by patrick-laa**
[Jira ticket](https://dsdmoj.atlassian.net/browse/EL-713)

## What changed and why

We now keep analytics in a postgres database.

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing
- Branch is generally up to date with main Github - definitely no conflicts
- No unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- PR description says *what* changed and *why*, with a link to the JIRA story.
- Diff has been checked for unexpected changes being included.
- Commit messages say why the change was made.

### CODE REVIEW

File: app/services/analytics_service.rb [hunk 1]
```diff
@@ -1,14 +1,12 @@
 class AnalyticsService
   class << self
     def call(event_type:, page:, assessment_code:, browser_id:)
-      hash = {
-        key: "CCQ Analytics Datum",
+      AnalyticsEvent.create(
```
Comment (starswan) [inline app/services/analytics_service.rb:4]: should this be create! (although if it fails, I'm not sure I want a hard crash...?)

File: db/migrate/20230221125618_create_analytics_events.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class CreateAnalyticsEvents < ActiveRecord::Migration[7.0]
+  def change
+    create_table :analytics_events do |t|
+      t.string :event_type
+      t.string :page
+      t.string :assessment_code
+      t.string :browser_id
```
Comment (starswan) [inline db/migrate/20230221125618_create_analytics_events.rb:7]: do we want some null: falses and some indexes in here?

File: app/services/analytics_service.rb (hunk 1 above)
Comment (patrick-laa) [reply app/services/analytics_service.rb:4]: Good idea! I'll add a `!` and then catch and Sentrify any exceptions generated.

File: db/migrate/20230221125618_create_analytics_events.rb (hunk 2 above)
Comment (patrick-laa) [reply db/migrate/20230221125618_create_analytics_events.rb:7]: Cool. I think the final say about the right indexes will come out of EL-717, but as a start I think it makes sense to index the created_at and assessment_code columns, as a lot of queries are based around those.

**Review by starswan — APPROVED** [review]

**Pull request merged**